src/main/scala/reporting/Converter.scala

```diff
@@ -329,26 +392,67 @@ object Converter {
   }
 
   def extractPredicate(chunk: BasicChunk, model: Model): HeapEntry = {
-    //this might be too simple for some cases but for prusti to tell if some
-    //variable is part of a class it should be good enough
-    //not really sure if the snap value should be added, seems to be same as one of
-    //the args in most cases.
+    // this might be too simple for some cases but for prusti to tell if some
+    // variable is part of a class it should be good enough
+    // not really sure if the snap value should be added, seems to be same as one of
+    // the args in most cases.
+    
     val argsEval = chunk.args.map(x => evaluateTerm(x, model))
-    val entry = PredHeapEntry(chunk.id.toString, argsEval)
-    entry
+    val perm: Option[Rational] = evalPerm(chunk.perm, model)
+    PredHeapEntry(chunk.id.toString, argsEval, perm)
   }
 
-  def extractSequence(
-      model: Model,
-      heap: ExtractedHeap,
-      nullRefName: String,
-      name: String,
-      elementSort: Sort,
-      encountered: Set[String]
-  ): ExtractedModelEntry = {
+  
+  def evalPerm(value: Term, model: Model): Option[Rational] = {
+    value match {      
+      case _: Var => evaluateTerm(value, model) match {
+        case LitPermEntry(value) => Some(value)
+        case _ => None
+      }
+      case App(_, _) => None
+      case NoPerm() => Some(Rational.zero)
+      case FullPerm() => Some(Rational.one)
+      case  FractionPermLiteral(r) => Some(r)
+      case _: FractionPerm => None
+      case IsValidPermVar(_) => None
+      case IsReadPermVar(_, _) => None
+      case PermTimes(v1, v2) => (evalPerm(v1, model), evalPerm(v2, model)) match {
+        case (Some(x), Some(y)) => Some(x * y)
+        case _ => None
+      }
```

Comment:
Maybe something like this for all of the operations?

```suggestion
      case PermTimes(v1, v2) =>
        evalPerm(v1, model).map(x => evalPerm(v2, model).map(y => x * y))
```